tests/unit/backend/chalice/api_server/test_v1_project.py

```diff
@@ -0,0 +1,154 @@
+import json
+import unittest
+from datetime import datetime
+
+from furl import furl
+
+from backend.corpora.common.corpora_orm import ProjectStatus
+from backend.corpora.common.entities import Project
+from tests.unit.backend.chalice.api_server import BaseAPITest
+from tests.unit.backend.utils import BogusProjectParams
+
+
+class TestProject(BaseAPITest, unittest.TestCase):
+    def test__list_project__ok(self):
+        path = "/v1/project"
+        headers = dict(host="localhost")
+
+        from_date = int(datetime.fromtimestamp(10).timestamp())
+        creation_time = 20
+        to_date = int(datetime.fromtimestamp(30).timestamp())
+
+        test_project = Project.create(
+            **BogusProjectParams.get(status=ProjectStatus.LIVE.name, created_at=datetime.fromtimestamp(creation_time)),
+        )
+        expected_id = test_project.id
+
+        with self.subTest("No Parameters"):
+            test_url = furl(path=path)
+            response = self.app.get(test_url.url, headers=headers)
+            response.raise_for_status()
+            actual_body = json.loads(response.body)
+            self.assertIn(expected_id, [p["id"] for p in actual_body["projects"]])
+            self.assertEqual(None, actual_body.get("to_date"))
+            self.assertEqual(None, actual_body.get("from_date"))
+
+        with self.subTest("from_date"):
+            test_url = furl(path=path, query_params={"from_date": from_date})
+            response = self.app.get(test_url.url, headers=headers)
+            response.raise_for_status()
+            actual_body = json.loads(response.body)
+            self.assertIn(expected_id, [p["id"] for p in actual_body["projects"]])
+            self.assertEqual(None, actual_body.get("to_date"))
+            self.assertEqual(actual_body["from_date"], from_date)
+
+        with self.subTest("to_date"):
+            test_url = furl(path=path, query_params={"to_date": to_date})
+            response = self.app.get(test_url.url, headers=headers)
+            response.raise_for_status()
+            actual_body = json.loads(response.body)
+            self.assertIn(expected_id, [p["id"] for p in actual_body["projects"]])
+            self.assertEqual(to_date, actual_body["to_date"])
+            self.assertEqual(None, actual_body.get("from_date"))
+
+        with self.subTest("from_date->to_date"):
+            test_url = furl(path=path, query_params={"from_date": from_date, "to_date": to_date})
+            response = self.app.get(test_url.url, headers=headers)
+            response.raise_for_status()
+            actual_body = json.loads(response.body)
+            self.assertEqual(expected_id, actual_body["projects"][0]["id"])
+            self.assertEqual(creation_time, actual_body["projects"][0]["created_at"])
+            self.assertEqual(from_date, actual_body["from_date"])
+            self.assertEqual(to_date, actual_body["to_date"])
+
+    @staticmethod
+    def remove_timestamps(body: dict) -> dict:
```

Comment:
Nit: move helper functions to the bottom of the file?

Reply:
fixed